.buildkite/quality-gates/queries/metrics_count.json

```diff
@@ -0,0 +1,63 @@
+{
+    "size": 0,
+    "query": {
+        "bool": {
+            "filter": [
+                {
+                    "term": {
+                        "kubernetes.deployment.name": "elasticsearch-metrics-apiserver"
+                    }
+                },
+                {
+                    "term": {
+                      "prometheus.labels.client": "k8s-observability-cluster"
+                    }
+                },
+                {
+                    "term": {
+                      "prometheus.labels.type": "custom"
+                    }
+                },
+                {
+                    "exists": {
+                        "field": "prometheus.metrics_count.value"
+                    }
+                },
+                {
+                    "range": {
+                        "@timestamp": {
+                            "gte": "now-10m"
+                        }
+                    }
+                }
+            ]
+        }
+    },
+    "aggs": {
+        "podNames":{
+            "terms": {
+                "size": 50,
+                "field": "kubernetes.pod.name"
+            },
+            "aggs":{
+              "latest_result": {
+                "top_hits": {
```

Comment:
IIUC we want to see that each adapter instance is reporting some metrics. I wonder if having a [stats](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-stats-aggregation.html) or [max](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-max-aggregation.html) agg here makes more sense, and we can just check that the max > 0. 

What we're actually checking is slightly different, but it would allow us the do the filtering entirely within the query via a [bucket selector](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-bucket-selector-aggregation.html) and the bash script would just be checking if any buckets were returned.

Reply:
@tobio thinking about this, we want to check every instance of the metrics-adapter pod has a metrics_count > 0, so in that case instead of max wouldn't we want to do a min and check that min is > 0 ?

Reply:
I think I see what you meant, we can do a max and filter out buckets that have max as 0